### ISSUE TITLE
Tests assume code is in a git repository

### ISSUE THREAD

**Issue by SammyTheSnake** [JuliaGPU/Vulkan.jl#44]
Running the tests seems to assume the package is in a git directory. If I do `]dev Vulkan`, I don't get this error message, so I'm guessing those working on Vulkan.jl just haven't spotted that it doesn't work in a standard (non-dev) install of the package...

I haven't looked into what the test is trying to do, but it seems that it should be working within Julia's packaging / artefact framework to do it, right? Downloading the whole development history as a git repository takes somewhat more time (especially if your internet connection is pants like mine) and space than just the version in use, so it'd be nice to do without that (and wouldn't it be nice if Julia supported using shallow repos - which would at least reduce the cost...)

I'm not sure if this is a bug in "Documenter", or a bug in how Vulkan.jl uses "Documenter", or a conscious decision for reasons outside my ken, but as something of a neophyte in the world of Julia, I'm hoping somebody who actually works on Vulkan.jl is able to determine that!

Note, this error log was generated using 1.9.0-rc3, but I get essentially the same using 1.8.5, so I don't think it's to do with the version of Julia I'm using.

```
JULIA_DEBUG=Documenter julia--project=.
] add Vulkan
(...)
] test Vulkan
(...)
┌ Warning: Unable to determine HTML(edit_link = ...) from remote HEAD branch, defaulting to "master".
│ Calling `git remote` failed with an exception. Set JULIA_DEBUG=Documenter to see the error.
│ Unless this is due to a configuration error, the relevant variable should be set explicitly.
└ @ Documenter.Utilities ~/.julia/packages/Documenter/H5y27/src/Utilities/Utilities.jl:822
┌ Debug: Command: setenv(`/usr/bin/git remote show origin`,[ ***ENV stripped for privacy/brevity*** ]; dir="/home/sammy/.julia/packages/Vulkan/76XsG/test")
│   exception =
│    failed process: Process(setenv(`/usr/bin/git remote show origin`,[ ***ENV stripped for privacy/brevity*** ]; dir="/home/sammy/.julia/packages/Vulkan/76XsG/test"), ProcessExited(128)) [128]
│
│    Stacktrace:
│      [1] pipeline_error
│        @ ./process.jl:565 [inlined]
│      [2] read(cmd::Base.CmdRedirect)
│        @ Base ./process.jl:449
│      [3] read
│        @ ./process.jl:458 [inlined]
│      [4] git_remote_head_branch(varname::String, root::String; remotename::String, fallback::String)
│        @ Documenter.Utilities ~/.julia/packages/Documenter/H5y27/src/Utilities/Utilities.jl:820
│      [5] git_remote_head_branch
│        @ ~/.julia/packages/Documenter/H5y27/src/Utilities/Utilities.jl:794 [inlined]
│      [6] Documenter.Writers.HTMLWriter.HTML()
│        @ Documenter.Writers.HTMLWriter ~/.julia/packages/Documenter/H5y27/src/Writers/HTMLWriter.jl:419
│      [7] (::Documenter.var"#all_doctests#32"{Bool, Vector{Regex}, Vector{Module}})()
│        @ Documenter ~/.julia/packages/Documenter/H5y27/src/Documenter.jl:942
│      [8] macro expansion
│        @ ~/opt/julia-1.9.0-rc3/share/julia/stdlib/v1.9/Test/src/Test.jl:478 [inlined]
│      [9] macro expansion
│        @ ~/.julia/packages/Documenter/H5y27/src/Documenter.jl:963 [inlined]
│     [10] macro expansion
│        @ ~/opt/julia-1.9.0-rc3/share/julia/stdlib/v1.9/Test/src/Test.jl:1498 [inlined]
│     [11] doctest(source::String, modules::Vector{Module}; fix::Bool, testset::String, doctestfilters::Vector{Regex})
│        @ Documenter ~/.julia/packages/Documenter/H5y27/src/Documenter.jl:963
│     [12] doctest
│        @ ~/.julia/packages/Documenter/H5y27/src/Documenter.jl:927 [inlined]
│     [13] doctest(package::Module; manual::Bool, testset::Nothing, kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
│        @ Documenter ~/.julia/packages/Documenter/H5y27/src/Documenter.jl:898
│     [14] doctest(package::Module)
│        @ Documenter ~/.julia/packages/Documenter/H5y27/src/Documenter.jl:884
│     [15] top-level scope
│        @ ~/.julia/packages/Vulkan/76XsG/test/base.jl:29
│     [16] include(mod::Module, _path::String)
│        @ Base ./Base.jl:457
│     [17] include(x::String)
│        @ Main.var"##292" ~/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:23
│     [18] macro expansion
│        @ ~/.julia/packages/Vulkan/76XsG/test/runtests.jl:4 [inlined]
│     [19] macro expansion
│        @ ~/opt/julia-1.9.0-rc3/share/julia/stdlib/v1.9/Test/src/Test.jl:1498 [inlined]
│     [20] top-level scope
│        @ ~/.julia/packages/Vulkan/76XsG/test/runtests.jl:4
│     [21] eval(m::Module, e::Any)
│        @ Core ./boot.jl:370
│     [22] top-level scope
│        @ ~/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:23
│     [23] include(fname::String)
│        @ Base.MainInclude ./client.jl:478
│     [24] top-level scope
│        @ none:6
│     [25] eval
│        @ ./boot.jl:370 [inlined]
│     [26] exec_options(opts::Base.JLOptions)
│        @ Base ./client.jl:280
│     [27] _start()
│        @ Base ./client.jl:522
│   stderr = "fatal: not a git repository (or any of the parent directories): .git\n"
└ @ Documenter.Utilities ~/.julia/packages/Documenter/H5y27/src/Utilities/Utilities.jl:827
(...)
```

As it happens, I'm probably not going to be able to use Vulkan.jl because my weenie GPU apparently only has support for vulkan 1.2, (perhaps that's a limitation of the underlying drivers and might change in future...) but I thought you might appreciate the feedback anyway :-)

**Comment by SammyTheSnake** [comment]
Not that it's super relevant, but it appears vulkan 1.3 support for my GPU is in the new drivers in debian testing, which I'll get in due course when bookworm goes stable, or somebody backports it, or if I decide to switch to testing...

**Comment by serenity4** [comment]
Hi,

Thanks for the feedback. You mention that you are getting an error, but all I get is a warning that has no incidence on testing. Did that impact the tests for you in any way?

**Comment by serenity4** [comment]
@SammyTheSnake any news here? Otherwise I'll close this as I can't replicate any errors and don't believe there is actually an issue.

**Comment by SammyTheSnake** [comment]
Sorry I didn't reply sooner (I actually only visit github intermittently!)

I mentioned in another post that my computer didn't have the relevant drivers for my GPU, so actually *running* the tests turned out to be a bit of a non-starter.

I'm not sure what warning you refer to, the error I get is in my original post.

If the (rather terminal sounding) error I got is spurious and can be ignored then perhaps it's just cosmetic, but still ought to be presented in a less terminal sounding way.

If the test is failing (which it clearly is for me, and in a way that doesn't appear to even get as far as a point where the drivers are implicated) then that doesn't seem to serve the purpose of tests very well. I figured it was a good way to see if this package does what I hope it might on my hardware and didn't enable me to answer that question (I managed to answer it myself ("no") by writing a mickey mouse program, though)

When you ran the tests, did you do it in a non-dev package installation? (i.e. `Pkg.add()` rather than `Pkd.dev()`)

Could it be that the specific versions of various related packages is relevant? I could post my manifest or something if that helps?

I've looked into upgrading my drivers but I'm using debian stable. This test is on a laptop that I'm a little reluctant to move to "testing" as I often go weeks between upgrading it (and my primary desktop happens to have an incompatible GPU) there doesn't seem to be a backported option for the relevant drivers, and trying to compile a backported version for myself rapidly turned into a rabbit warren of "now re-compile *this* dependency..." that I didn't fancy following too far...

**Comment by serenity4** [comment]
> When you ran the tests, did you do it in a non-dev package installation? (i.e. Pkg.add() rather than Pkd.dev())

Yes, and I only had a minor warning during the tests.

The error that you mention (`failed process: Process(setenv(`/usr/bin/git remote show origin`) is just that Documenter tries to get the address of a remote, but it doesn't work since there is no Git repo, and then it just proceeds normally (at least when I do it locally).

Could you post the full output of running the tests, from `test Vulkan` to the test report? So far I only see warnings or debug logs.

**Comment by SammyTheSnake** [comment]
I've just re-done the whole thing from scratch in a temp project with no other packages installed (other than dependencies) in both 1.8.5 and 1.9.0-rc3 and didn't get the same errors, indeed, it looks like all the tests pass (though there seem to be more tests when run in 1.9, which I didn't expect...)

I wonder if in my previous run there was some other package installed that somehow triggered the Documenter behaviour...?

It could be that there's a dependency interaction somewhere with one of the other packages I was using that hits an incompatibility not fully expressed in Vulkan.jl's dependency specification?

I also noticed something else that *might* be tangentially related: when running the tests under 1.8 (but not 1.9) that I got a warning about needing to re-resolve the dependencies, even with only Vulkan.jl installed, which is odd, too. Does the test require additional dependencies not used in the non-test case, that then have different version requirements for some of the dependencies? Again, if the tests don't use the same versions of packages/dependencies as the actual versions used, then that works against the applicability of the test results...

```
(jl_EdnPr0) pkg> test Vulkan
     Testing Vulkan
┌ Warning: Could not use exact versions of packages in manifest, re-resolving
└ @ Pkg.Operations ~/src/opt/julia/usr/share/julia/stdlib/v1.8/Pkg/src/Operations.jl:1601
```

Whatever these potential dependency oddities are, it's likely to be a fairly subtle thing to track down, so I'm inclined to say this bug should be closed and if I hit something similar at some future point when I've upgraded my drivers and start actually trying to *use* Vulkan.jl (thus meaning the "clean install ignoring other packages" approach doesn't apply) then I'll create a new issue to track it (and maybe even identify exactly where in the dependency graph things go squiffy - if I've got my head around the dependency graph a little more!)

If you want to follow the thread a little more, I'd be happy to try to reproduce the error I got before and post the full output as you mentioned before. Incidentally, I put in the "DEBUG=Documenter" bit because that was suggested in the error message I initially got, and gave some more information that seemed relevant, it isn't what *triggered* the error...

Anyway, thanks for taking the time to write Vulkan.jl and to read my issue report :-)

**Comment by serenity4** [comment]
There will be fewer tests in versions < 1.9 because a few of them are related to package extensions which are new in 1.9.

I just noticed I recently removed doctests by accident, running tests against the current `master` branch should have warnings back.

About the re-resolving behavior, I don't know what causes this, and I don't think this is harmful. You could post an issue in Pkg.jl about it.

This is what I get when running tests after `Pkg.add`:

<details>
  <summary>Result of running tests</summary><p>

```julia
(@v1.8) pkg> activate --temp
  Activating new project at `/tmp/jl_BueKX6`

(jl_BueKX6) pkg> add Vulkan#master
    Updating git-repo `https://github.com/JuliaGPU/Vulkan.jl.git`
test     Updating registry at `~/.julia/registries/Graphics`
    Updating git-repo `git@github.com:serenity4/Graphics`
Vulkan
    Updating registry at `~/.julia/registries/General.toml`
   Resolving package versions...
    Updating `/tmp/jl_BueKX6/Project.toml`
  [9f14b124] + Vulkan v0.6.6 `https://github.com/JuliaGPU/Vulkan.jl.git#master`
    Updating `/tmp/jl_BueKX6/Manifest.toml`
  [7d9f7c33] + Accessors v0.1.30
  [15f4f7f2] + AutoHashEquals v0.2.0
  [a3e06817] + BitMasks v0.1.0
  [34da2185] + Compat v4.6.1
  [a33af91c] + CompositionsBase v0.1.1
  [187b0558] + ConstructionBase v1.5.2
  [ffbed154] + DocStringExtensions v0.9.3
  [3587e190] + InverseFunctions v0.1.9
  [692b3bcd] + JLLWrappers v1.4.1
  [d8e11817] + MLStyle v0.4.17
  [1914dd2f] + MacroTools v0.5.10
  [aea7be01] + PrecompileTools v1.1.0
  [21216c6a] + Preferences v1.4.0
  [189a3867] + Reexport v1.2.2
  [ae029012] + Requires v1.3.0
  [08a2b407] + ResultTypes v3.1.0
  [90137ffa] + StaticArrays v1.5.24
  [1e83bf80] + StaticArraysCore v1.4.0
  [9f14b124] + Vulkan v0.6.6 `https://github.com/JuliaGPU/Vulkan.jl.git#master`
  [16167f82] + VulkanCore v1.3.0
  [8d446b21] + Vulkan_Headers_jll v1.3.240+0
  [0dad84c5] + ArgTools v1.1.1
  [56f22d72] + Artifacts
  [2a0f44e3] + Base64
  [ade2ca70] + Dates
  [f43a241f] + Downloads v1.6.0
  [7b1f6079] + FileWatching
  [b77e0a4c] + InteractiveUtils
  [b27032c2] + LibCURL v0.6.3
  [76f85450] + LibGit2
  [8f399da3] + Libdl
  [37e2e46d] + LinearAlgebra
  [56ddb016] + Logging
  [d6f4376e] + Markdown
  [ca575930] + NetworkOptions v1.2.0
  [44cfe95a] + Pkg v1.8.0
  [de0858da] + Printf
  [3fa0cd96] + REPL
  [9a3f8284] + Random
  [ea8e919c] + SHA v0.7.0
  [9e88b42a] + Serialization
  [6462fe0b] + Sockets
  [2f01184e] + SparseArrays
  [10745b16] + Statistics
  [fa267f1f] + TOML v1.0.0
  [a4e569a6] + Tar v1.10.1
  [8dfed614] + Test
  [cf7118a7] + UUIDs
  [4ec0a83e] + Unicode
  [e66e0078] + CompilerSupportLibraries_jll v1.0.1+0
  [deac9b47] + LibCURL_jll v7.84.0+0
  [29816b5a] + LibSSH2_jll v1.10.2+0
  [c8ffd9c3] + MbedTLS_jll v2.28.0+0
  [14a3606d] + MozillaCACerts_jll v2022.2.1
  [4536629a] + OpenBLAS_jll v0.3.20+0
  [83775a58] + Zlib_jll v1.2.12+3
  [8e850b90] + libblastrampoline_jll v5.1.1+0
  [8e850ede] + nghttp2_jll v1.48.0+0
  [3f19e933] + p7zip_jll v17.4.0+0
Precompiling project...
  1 dependency successfully precompiled in 43 seconds. 26 already precompiled.

(jl_BueKX6) pkg> test Vulkan
     Testing Vulkan
┌ Warning: Could not use exact versions of packages in manifest, re-resolving
└ @ Pkg.Operations /cache/build/default-amdci4-2/julialang/julia-release-1-dot-8/usr/share/julia/stdlib/v1.8/Pkg/src/Operations.jl:1601
      Status `/tmp/jl_iIc6A2/Project.toml`
  [7d9f7c33] Accessors v0.1.30
  [15f4f7f2] AutoHashEquals v0.2.0
  [a3e06817] BitMasks v0.1.0
  [3da002f7] ColorTypes v0.11.4
⌅ [ffbed154] DocStringExtensions v0.8.6
  [e30172f5] Documenter v0.27.24
  [d8e11817] MLStyle v0.4.17
  [aea7be01] PrecompileTools v1.1.0
  [21216c6a] Preferences v1.4.0
  [189a3867] Reexport v1.2.2
  [08a2b407] ResultTypes v3.1.0
  [9f14b124] Vulkan v0.6.6 `https://github.com/JuliaGPU/Vulkan.jl.git#master`
  [16167f82] VulkanCore v1.3.0
  [16167f82] XCB v0.5.0
  [c404ff5a] SwiftShader_jll v0.1.0+0
  [8f399da3] Libdl `@stdlib/Libdl`
  [56ddb016] Logging `@stdlib/Logging`
  [44cfe95a] Pkg v1.8.0 `@stdlib/Pkg`
  [8dfed614] Test `@stdlib/Test`
      Status `/tmp/jl_iIc6A2/Manifest.toml`
  [a4c015fc] ANSIColoredPrinters v0.0.1
  [7d9f7c33] Accessors v0.1.30
  [15f4f7f2] AutoHashEquals v0.2.0
  [a3e06817] BitMasks v0.1.0
  [fa961155] CEnum v0.4.2
  [3da002f7] ColorTypes v0.11.4
  [34da2185] Compat v4.6.1
  [a33af91c] CompositionsBase v0.1.1
  [187b0558] ConstructionBase v1.5.2
⌅ [ffbed154] DocStringExtensions v0.8.6
  [e30172f5] Documenter v0.27.24
  [53c48c17] FixedPointNumbers v0.8.4
  [b5f81e59] IOCapture v0.2.2
  [3587e190] InverseFunctions v0.1.9
  [692b3bcd] JLLWrappers v1.4.1
  [682c06a0] JSON v0.21.4
  [d8e11817] MLStyle v0.4.17
  [1914dd2f] MacroTools v0.5.10
  [69de0a69] Parsers v2.5.8
  [aea7be01] PrecompileTools v1.1.0
  [21216c6a] Preferences v1.4.0
  [189a3867] Reexport v1.2.2
  [ae029012] Requires v1.3.0
  [08a2b407] ResultTypes v3.1.0
  [66db9d55] SnoopPrecompile v1.0.3
  [90137ffa] StaticArrays v1.5.24
  [1e83bf80] StaticArraysCore v1.4.0
  [3a884ed6] UnPack v1.0.2
  [9f14b124] Vulkan v0.6.6 `https://github.com/JuliaGPU/Vulkan.jl.git#master`
  [16167f82] VulkanCore v1.3.0
  [e18202ca] WindowAbstractions v0.5.0
  [16167f82] XCB v0.5.0
  [2e619515] Expat_jll v2.4.8+0
⌅ [e9f186c6] Libffi_jll v3.2.2+1
  [d4300ac3] Libgcrypt_jll v1.8.7+0
  [7add5ba3] Libgpg_error_jll v1.42.0+0
  [94ce4f54] Libiconv_jll v1.16.1+2
  [c404ff5a] SwiftShader_jll v0.1.0+0
  [8d446b21] Vulkan_Headers_jll v1.3.240+0
  [a2964d1f] Wayland_jll v1.21.0+0
  [2381bf8a] Wayland_protocols_jll v1.25.0+0
  [02c8fc9c] XML2_jll v2.10.3+0
  [aed1982a] XSLT_jll v1.1.34+0
  [4f6342f7] Xorg_libX11_jll v1.6.9+4
  [0c0b7dd1] Xorg_libXau_jll v1.0.9+4
  [a3789734] Xorg_libXdmcp_jll v1.1.3+4
  [14d82f49] Xorg_libpthread_stubs_jll v0.1.0+3
  [c7cfdc94] Xorg_libxcb_jll v1.13.0+3
  [cc61e674] Xorg_libxkbfile_jll v1.1.0+4
  [2def613f] Xorg_xcb_util_jll v0.4.0+1
  [35661453] Xorg_xkbcomp_jll v1.4.2+4
  [33bec58e] Xorg_xkeyboard_config_jll v2.27.0+4
  [c5fb5394] Xorg_xtrans_jll v1.4.0+3
⌅ [d8fb68d0] xkbcommon_jll v0.9.1+5
  [0dad84c5] ArgTools v1.1.1 `@stdlib/ArgTools`
  [56f22d72] Artifacts `@stdlib/Artifacts`
  [2a0f44e3] Base64 `@stdlib/Base64`
  [ade2ca70] Dates `@stdlib/Dates`
  [f43a241f] Downloads v1.6.0 `@stdlib/Downloads`
  [7b1f6079] FileWatching `@stdlib/FileWatching`
  [b77e0a4c] InteractiveUtils `@stdlib/InteractiveUtils`
  [b27032c2] LibCURL v0.6.3 `@stdlib/LibCURL`
  [76f85450] LibGit2 `@stdlib/LibGit2`
  [8f399da3] Libdl `@stdlib/Libdl`
  [37e2e46d] LinearAlgebra `@stdlib/LinearAlgebra`
  [56ddb016] Logging `@stdlib/Logging`
  [d6f4376e] Markdown `@stdlib/Markdown`
  [a63ad114] Mmap `@stdlib/Mmap`
  [ca575930] NetworkOptions v1.2.0 `@stdlib/NetworkOptions`
  [44cfe95a] Pkg v1.8.0 `@stdlib/Pkg`
  [de0858da] Printf `@stdlib/Printf`
  [3fa0cd96] REPL `@stdlib/REPL`
  [9a3f8284] Random `@stdlib/Random`
  [ea8e919c] SHA v0.7.0 `@stdlib/SHA`
  [9e88b42a] Serialization `@stdlib/Serialization`
  [6462fe0b] Sockets `@stdlib/Sockets`
  [2f01184e] SparseArrays `@stdlib/SparseArrays`
  [10745b16] Statistics `@stdlib/Statistics`
  [fa267f1f] TOML v1.0.0 `@stdlib/TOML`
  [a4e569a6] Tar v1.10.1 `@stdlib/Tar`
  [8dfed614] Test `@stdlib/Test`
  [cf7118a7] UUIDs `@stdlib/UUIDs`
  [4ec0a83e] Unicode `@stdlib/Unicode`
  [e66e0078] CompilerSupportLibraries_jll v1.0.1+0 `@stdlib/CompilerSupportLibraries_jll`
  [deac9b47] LibCURL_jll v7.84.0+0 `@stdlib/LibCURL_jll`
  [29816b5a] LibSSH2_jll v1.10.2+0 `@stdlib/LibSSH2_jll`
  [c8ffd9c3] MbedTLS_jll v2.28.0+0 `@stdlib/MbedTLS_jll`
  [14a3606d] MozillaCACerts_jll v2022.2.1 `@stdlib/MozillaCACerts_jll`
  [4536629a] OpenBLAS_jll v0.3.20+0 `@stdlib/OpenBLAS_jll`
  [83775a58] Zlib_jll v1.2.12+3 `@stdlib/Zlib_jll`
  [8e850b90] libblastrampoline_jll v5.1.1+0 `@stdlib/libblastrampoline_jll`
  [8e850ede] nghttp2_jll v1.48.0+0 `@stdlib/nghttp2_jll`
  [3f19e933] p7zip_jll v17.4.0+0 `@stdlib/p7zip_jll`
        Info Packages marked with ⌅ have new versions available but compatibility constraints restrict them from upgrading.
     Testing Running tests...
[ Info: Instance created: Instance(Ptr{Nothing} @0x000000000296bfd0).
[ Info: General (Loader Message): linux_read_sorted_physical_devices:
[ Info: General (Loader Message):      Original order:
[ Info: General (Loader Message):            [0] SwiftShader Device (LLVM 10.0.0)
[ Info: General (Loader Message):      Sorted order:
[ Info: General (Loader Message):            [0] SwiftShader Device (LLVM 10.0.0)  
[ Info: General (Loader Message): linux_read_sorted_physical_devices:
[ Info: General (Loader Message):      Original order:
[ Info: General (Loader Message):            [0] SwiftShader Device (LLVM 10.0.0)
[ Info: General (Loader Message):      Sorted order:
[ Info: General (Loader Message):            [0] SwiftShader Device (LLVM 10.0.0)  
[ Info: 1 physical device(s) found.
[ Info: Selected SwiftShader Device (LLVM 10.0.0) - PHYSICAL_DEVICE_TYPE_CPU (driver: 5.0.0, supported Vulkan API: 1.3.0)
[ Info: General (Loader Message): Inserted device layer "VK_LAYER_KHRONOS_validation" (libVkLayer_khronos_validation.so)
[ Info: General (Loader Message):        Using "SwiftShader Device (LLVM 10.0.0)" with driver: "/home/belmant/.julia/artifacts/92ef6528ec0492010323f96b079eb94fd838b31e/lib/./libvk_swiftshader.so"
[ Info: Logical device created: Device(Ptr{Nothing} @0x0000000002806860)
┌ Warning: Unable to determine HTML(edit_link = ...) from remote HEAD branch, defaulting to "master".
│ Calling `git remote` failed with an exception. Set JULIA_DEBUG=Documenter to see the error.
│ Unless this is due to a configuration error, the relevant variable should be set explicitly.
└ @ Documenter.Utilities ~/.julia/packages/Documenter/H5y27/src/Utilities/Utilities.jl:822
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: running doctests.
[ Info: Skipped ExpandTemplates step (doctest only).
[ Info: Skipped CrossReferences step (doctest only).
[ Info: Skipped CheckDocument step (doctest only).
[ Info: Skipped Populate step (doctest only).
[ Info: Skipped RenderDocument step (doctest only).
Test Summary: | Pass  Total   Time
Vulkan.jl     |   50     50  44.7s
     Testing Vulkan tests passed 

(jl_BueKX6) pkg> 
```

</p> </details>

To me it seems like everything is working fine, all tests pass and no errors are thrown. The warning comes from Documenter.jl and is harmless for the purposes of testing. Feel free to make an issue in the Documenter repo if the warning troubles you, otherwise you can safely ignore it.

I'll close this issue as the items discussed there are out of scope of this package and are specific to Pkg.jl and Documenter.jl.

**Comment by SammyTheSnake** [comment]
I've done some further poking and have made some observations.

TL;DR: I think the issue seems to only happen in Vulkan.jl version 0.6.4, but not in the newer 0.6.6 - on that basis, I think that closing this issue is probably fair.

It looks like I ended up with Vulkan.jl version 0.6.4 with the other packages installed, but 0.6.6 when done in a clean installation, so there's probably some subtle dependency version requirement interaction involved. I have a suspicion that there might be some bugs to find in the dependency tree somewhere, such as under-specified version requirements, but it's at least as likely to be in one of the other packages, so let's assume that for now :-P

The clean installation ran with only non-fatal sounding warning messages, but if I specified Vulkan.jl 0.6.4, I get the same errors as before (note, the initial test includes a less verbose error message that suggests adding `JULIA_DEBUG=Documenter` which is why that's there)

I've attached four session logs (Julia 1.8.5 & 1.9.0-rc3, and Vulkan 0.6.4 & 0.6.6) - all from clean non-dev installs, the ones with the Documenter error were run with `JULIA_DEBUG=Documenter`, but a terser version of the same error appeared without that.

[Julia-1.8.5_Vulkan-0.6.4.txt](https://github.com/JuliaGPU/Vulkan.jl/files/11421638/Julia-1.8.5_Vulkan-0.6.4.txt)
[Julia-1.8.5_Vulkan-0.6.6.txt](https://github.com/JuliaGPU/Vulkan.jl/files/11421639/Julia-1.8.5_Vulkan-0.6.6.txt)
[Julia-1.9.0-rc3_Vulkan-0.6.4.txt](https://github.com/JuliaGPU/Vulkan.jl/files/11421641/Julia-1.9.0-rc3_Vulkan-0.6.4.txt)
[Julia-1.9.0-rc3_Vulkan-0.6.6.txt](https://github.com/JuliaGPU/Vulkan.jl/files/11421642/Julia-1.9.0-rc3_Vulkan-0.6.6.txt)

**Comment by serenity4** [comment]
Thanks for the logs.

The error on 0.6.4 occurred because of this: `PrecompilationError: No physical device was found which supports Vulkan 1.3 or higher.`

I changed to use the SwiftShader driver yesterday for local testing (instead of trying to use system drivers) which made that error disappear in 0.6.6. Note that 0.6.5 and 0.6.6 were released yesterday, which explains why you ended up with 0.6.4 a few days ago (when it was indeed the latest version).
